### DISCUSSION TITLE
Ruby installation failed in macOS Big Sur

### DISCUSSION THREAD

**Discussion by balrajOla** [rbenv/ruby-build#1937] [Build failures]
Unable to install any ruby version and budler on macOS Big Sur

I am following these steps:-
```brew install rbenv
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/cask).
==> Updated Casks
Updated 3 casks.

==> Downloading https://homebrew.bintray.com/bottles/rbenv-1.1.2.big_sur.bottle.tar.gz
Already downloaded: /Users/name/Library/Caches/Homebrew/downloads/dbb4da662766f6b1e426b189d6d9a17d063eb9691bbbe62b78e832f82900350e--rbenv-1.1.2.big_sur.bottle.tar.gz
==> Pouring rbenv-1.1.2.big_sur.bottle.tar.gz
🍺  /usr/local/Cellar/rbenv/1.1.2: 36 files, 105.2KB

rbenv install
Downloading openssl-1.1.1i.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/e8be6a35fe41d10603c3cc635e93289ed00bf34b79671a3a4de64fcee00d5242
Installing openssl-1.1.1i...
Installed openssl-1.1.1i to /Users/name/.rbenv/versions/2.6.5

Downloading ruby-2.6.5.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.5.tar.bz2
Installing ruby-2.6.5...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.2.1 using ruby-build 20210119)

Inspect or clean up the working tree at /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216124622.91669.v6m2Eq
Results logged to /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216124622.91669.log

Last 10 log lines:
ld: warning: dylib (/usr/local/lib/libgmp.dylib) was built for newer macOS version (11.0) than being linked (10.15.6)
Undefined symbols for architecture x86_64:
  "__mh_execute_header", referenced from:
      _rb_dump_backtrace_with_lines in addr2line.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [libruby.2.6.dylib] Error 1
make: *** Waiting for unfinished jobs....
ln -sf ../../../.ext/include/x86_64-darwin20/rb_mjit_min_header-2.6.5.h include/ruby-2.6.0/x86_64-darwin20/rb_mjit_min_header-2.6.5.h
transdb.h updated
name-Singh:number26_ios name$ brew install openssl
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 1 formula.

Warning: openssl@1.1 1.1.1i is already installed and up-to-date.
To reinstall 1.1.1i, run:
  brew reinstall openssl@1.1

$ export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)"

$ rbenv install 2.7.0

Downloading ruby-2.7.0.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.0.tar.bz2
Installing ruby-2.7.0...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.2.1 using ruby-build 20210119)

Inspect or clean up the working tree at /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216125938.15886.AZCdH0
Results logged to /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216125938.15886.log

Last 10 log lines:
    VALUE mesg;
              ^
               = 0
eval.c:744:5: warning: function 'rb_f_raise' declared 'noreturn' should not return [-Winvalid-noreturn]
    UNREACHABLE_RETURN(Qnil);
    ^
./include/ruby/ruby.h:59:35: note: expanded from macro 'UNREACHABLE_RETURN'
#  define UNREACHABLE_RETURN(val) return (val)
                                  ^
3 warnings generated.
```

I am unable to install any version of ruby. Please help.

**Comment by ynelin** [comment]
Had the same problem on BigSur

Reinstalled OpenSSL, rbenv, ruby-build, llvm and it helps

**Comment by hsbt** [comment]
I never reproduce it. If you provide the instructions for reproducing this, I can help you.

**Comment by premshiva143** [comment]
> Unable to install any ruby version and budler on macOS Big Sur
> 
> I am following these steps:-
> 
> ```
> Updating Homebrew...
> ==> Auto-updated Homebrew!
> Updated 1 tap (homebrew/cask).
> ==> Updated Casks
> Updated 3 casks.
> 
> ==> Downloading https://homebrew.bintray.com/bottles/rbenv-1.1.2.big_sur.bottle.tar.gz
> Already downloaded: /Users/name/Library/Caches/Homebrew/downloads/dbb4da662766f6b1e426b189d6d9a17d063eb9691bbbe62b78e832f82900350e--rbenv-1.1.2.big_sur.bottle.tar.gz
> ==> Pouring rbenv-1.1.2.big_sur.bottle.tar.gz
> 🍺  /usr/local/Cellar/rbenv/1.1.2: 36 files, 105.2KB
> 
> rbenv install
> Downloading openssl-1.1.1i.tar.gz...
> -> https://dqw8nmjcqpjn7.cloudfront.net/e8be6a35fe41d10603c3cc635e93289ed00bf34b79671a3a4de64fcee00d5242
> Installing openssl-1.1.1i...
> Installed openssl-1.1.1i to /Users/name/.rbenv/versions/2.6.5
> 
> Downloading ruby-2.6.5.tar.bz2...
> -> https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.5.tar.bz2
> Installing ruby-2.6.5...
> ruby-build: using readline from homebrew
> 
> BUILD FAILED (macOS 11.2.1 using ruby-build 20210119)
> 
> Inspect or clean up the working tree at /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216124622.91669.v6m2Eq
> Results logged to /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216124622.91669.log
> 
> Last 10 log lines:
> ld: warning: dylib (/usr/local/lib/libgmp.dylib) was built for newer macOS version (11.0) than being linked (10.15.6)
> Undefined symbols for architecture x86_64:
>   "__mh_execute_header", referenced from:
>       _rb_dump_backtrace_with_lines in addr2line.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make: *** [libruby.2.6.dylib] Error 1
> make: *** Waiting for unfinished jobs....
> ln -sf ../../../.ext/include/x86_64-darwin20/rb_mjit_min_header-2.6.5.h include/ruby-2.6.0/x86_64-darwin20/rb_mjit_min_header-2.6.5.h
> transdb.h updated
> name-Singh:number26_ios name$ brew install openssl
> Updating Homebrew...
> ==> Auto-updated Homebrew!
> Updated 1 tap (homebrew/core).
> ==> Updated Formulae
> Updated 1 formula.
> 
> Warning: openssl@1.1 1.1.1i is already installed and up-to-date.
> To reinstall 1.1.1i, run:
>   brew reinstall openssl@1.1
> 
> $ export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)"
> 
> $ rbenv install 2.7.0
> 
> Downloading ruby-2.7.0.tar.bz2...
> -> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.0.tar.bz2
> Installing ruby-2.7.0...
> ruby-build: using readline from homebrew
> 
> BUILD FAILED (macOS 11.2.1 using ruby-build 20210119)
> 
> Inspect or clean up the working tree at /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216125938.15886.AZCdH0
> Results logged to /var/folders/0m/d_d_3jb10hnbqhrwf01p5v380000gp/T/ruby-build.20210216125938.15886.log
> 
> Last 10 log lines:
>     VALUE mesg;
>               ^
>                = 0
> eval.c:744:5: warning: function 'rb_f_raise' declared 'noreturn' should not return [-Winvalid-noreturn]
>     UNREACHABLE_RETURN(Qnil);
>     ^
> ./include/ruby/ruby.h:59:35: note: expanded from macro 'UNREACHABLE_RETURN'
> #  define UNREACHABLE_RETURN(val) return (val)
>                                   ^
> 3 warnings generated.
> ```
> 
> I am unable to install any version of ruby. Please help.

any luck, im also facing same issue?

**Comment by premshiva143** [comment]
> I never reproduce it. If you provide the instructions for reproducing this, I can help you.

Which version of mac os you are using?

**Comment by gplusdotgr** [comment]
same issue

**Comment by sidewaysdesign** [comment]
I'm also experiencing this exact issue, under MacOS Monterey.

**Comment by MuhammedZatar** [comment]
Has anyone found a fix for this? running into too. Running MacOS Monterey (12.2.1).

**Comment by mohanzhang** [comment]
I ran into this today and ended up fixing it through one or more of these steps (sorry, this wasn't a very well-controlled experiment):
1. `brew update`
2. `brew upgrade`
3. (rbenv install still wasn't working at this point)
4. Went nuclear and reinstalled xcode developer tools (this will take a while): https://apple.stackexchange.com/questions/93573/how-to-reinstall-xcode-command-line-tools
5. `brew uninstall --ignore-dependencies ruby-build`
6. `brew install ruby-build`
7. `rbenv install 2.7.6` <-- suddenly works!

Again, not sure which of these steps is strictly necessary, but I have got past this issue and figured I'd share it here in case it helps anyone else.
